js/components/lab-model.js

```diff
@@ -20,9 +20,7 @@ export default class LabModel extends PureComponent {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.model !== prevProps.model) {
-      this.setState({loading: true});
-    }
+    this.setState({ loading: (this.props.model !== prevProps.model) });
```

Comment:
Good advice on state, thanks! I was trying to fix the never ending circle and thought it might have been due to the forever setting of loading=true. As it is, with the external sensor I'm getting that circle to appear and never go away unless I turn it off somehow, so I guess I'll try a different approach again. I did also wonder if it was related to continual http requests in the background, but the circular progress indicator is different from the refresh indicator, so I don't think that's the root cause.